source/text_handler.cpp

```diff
@@ -229,25 +229,27 @@ spv_result_t AssemblyContext::binaryEncodeU64(const uint64_t value,
 spv_result_t AssemblyContext::binaryEncodeNumericLiteral(
     const char* val, spv_result_t error_code, const IdType& type,
     spv_instruction_t* pInst) {
-  const bool is_bottom = type.type_class == libspirv::IdTypeClass::kBottom;
-  const bool is_floating = libspirv::isScalarFloating(type);
-  const bool is_integer = libspirv::isScalarIntegral(type);
-
-  if (!is_bottom && !is_floating && !is_integer) {
-    return diagnostic(SPV_ERROR_INTERNAL)
-           << "The expected type is not a scalar integer or float type";
+  using spvutils::EncodeNumberStatus;
+  EncodeNumberStatus rc = EncodeNumberStatus::kSuccess;
+  std::string error_msg;
+  std::tie(rc, error_msg) = ParseAndEncodeNumber(
+      val, type,
+      [this, pInst](uint32_t d) { this->binaryEncodeU32(d, pInst); });
+  switch (rc) {
+    case EncodeNumberStatus::kSuccess:
+      return SPV_SUCCESS;
+    case EncodeNumberStatus::kInvalidText:
+      // Number value does not fit the bit width of the expected type (IdType)
```

Comment:
I think it's better to put these comments in the definition of `EncodeNumberStatus`.